ksqldb-execution/src/main/java/io/confluent/ksql/execution/codegen/SqlToJavaVisitor.java

```diff
@@ -360,15 +364,32 @@ private String formatExpression(final Expression expression) {
     }
 
     @Override
+    // CHECKSTYLE_RULES.OFF: TodoComment
     public Pair<String, SqlType> visitLambdaExpression(
         final LambdaFunctionCall lambdaFunctionCall, final TypeContext context) {
-      return visitUnsupported(lambdaFunctionCall);
+
+      context.mapLambdaInputTypes(lambdaFunctionCall.getArguments());
+      final Pair<String, SqlType> lambdaBody = process(lambdaFunctionCall.getBody(), context);
+
+      final List<Pair<String, Class<?>>> argPairs = new ArrayList<>();
+      for (final String lambdaArg: lambdaFunctionCall.getArguments()) {
+        argPairs.add(new Pair<>(
+            lambdaArg,
+            SchemaConverters.sqlToJavaConverter().toJavaType(context.getLambdaType(lambdaArg))
+            ));
+      }
+      return new Pair<>(LambdaUtil.function(argPairs, lambdaBody.getLeft()),
```

Comment:
```suggestion
      return new Pair<>(LambdaUtil.toJavaCode(argPairs, lambdaBody.getLeft()),
```
let's name this method something more meaningful